posthog/temporal/batch_exports/snowflake_batch_export.py

```diff
@@ -147,28 +160,70 @@ class SnowflakeClient:
     """Snowflake connection client used in batch exports."""
 
     def __init__(
-        self, user: str, password: str, account: str, warehouse: str, database: str, schema: str, role: str | None
+        self,
+        user: str,
+        account: str,
+        warehouse: str,
+        database: str,
+        schema: str,
+        role: str | None = None,
+        password: str | None = None,
+        private_key_file: str | None = None,
+        private_key_file_pwd: bytes | None = None,
     ):
+        if password is None and private_key_file is None:
+            raise ValueError("Either password or private key must be provided")
+
+        self.role = role
         self.user = user
         self.password = password
+        self.private_key_file = private_key_file
+        self.private_key_file_pwd = private_key_file_pwd
         self.account = account
         self.warehouse = warehouse
         self.database = database
         self.schema = schema
-        self.role = role
         self._connection: SnowflakeConnection | None = None
 
     @classmethod
     def from_inputs(cls, inputs: SnowflakeInsertInputs) -> typing.Self:
         """Initialize `SnowflakeClient` from `SnowflakeInsertInputs`."""
+
+        # User could have specified both password and private key in their batch export config.
+        # (for example, if they've already created a batch export with password auth and are now switching to keypair auth)
+        # Therefore we decide which one to use based on the authentication_type.
+        password = None
+        private_key_file_path = None
+        private_key_file_pwd = None
+        if inputs.authentication_type == "password":
+            password = inputs.password
+            if password is None:
+                raise ValueError("Password is required for password authentication")
```

Comment:
Should be made non-retryable or re-raised as non-retryable

Reply:
yes you're right. I had just added this as a sanity check since in theory it should never be the case as the http layer prevents a password not being yet. But does make sense for it to be made non-retryable just in case it does happen